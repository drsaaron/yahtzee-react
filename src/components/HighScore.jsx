import { useEffect } from 'react';

const zeroPad = (num, places) => String(num).padStart(places, '0');

function formatDate(d) {
    var date = d.getDate();
    var month = d.getMonth() + 1;
    var year = d.getFullYear();

    return year.toString() + "-" + zeroPad(month, 2) + "-" + zeroPad(date, 2);
}

const HighScore = (props) => {

    useEffect(() => {
	props.getHighScore();
    }, []);

    var scoreDate = props.highScore.date;
    var scoreDateTxt = '';
    if (scoreDate > 0) {
	scoreDateTxt = formatDate((new Date(scoreDate)));
    }

    return (
	<div id="highScore">
	    High Score: {props.highScore.highScore} ({props.highScore.name} on {scoreDateTxt})
	</div>
    );
};

export default HighScore;
