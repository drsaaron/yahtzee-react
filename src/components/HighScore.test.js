import React from 'react';
import renderer from 'react-test-renderer';

import HighScore from './HighScore';

describe('highScore', () => {
    test('basic', () => {
	var highScore = { highScore: 250, date: 1644810598778 }; // this date is 2/14/2022 in GMT but 2/13/2022 in local time
	const getHighScore = () => { };
	const component = renderer.create(<HighScore highScore={highScore} getHighScore={getHighScore} />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
    });
});
