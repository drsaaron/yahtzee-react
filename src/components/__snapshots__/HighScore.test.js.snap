// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highScore basic 1`] = `
<div
  id="highScore"
>
  High Score: 
  250
   (
   on 
  2022-02-13
   )
</div>
`;
